.buildkite/generator.jl

```diff
@@ -0,0 +1,229 @@
+const DEBUG = !haskey(ENV, "BUILDKITE")
+
+include("utils.jl")
+
+# Normally we look at the last pushed commit
+COMPARE_AGAINST="HEAD~1"
+
+# Keyword to be used in the commit message to skip a rebuild
+# TODO: Not run generator? Add to bk pipeline directly
+const SKIP_BUILD_COOKIE="[skip build]"
+
+# This variable will tell us whether we want to skip the build
+SKIP_BUILD=false
+
+const IS_PR = get(ENV, "BUILDKITE_PULL_REQUEST", "false") != "false"
+
+exec(cmd) = @assert success(pipeline(cmd, stderr=stderr, stdout=stdout))
+
+if IS_PR
+    # If we're on a PR though, we look at the entire branch at once
+    BASE_BRANCH = get(ENV, "BUILDKITE_PULL_REQUEST_BASE_BRANCH", "master")
+    @assert !isempty(BASE_BRANCH)
+
+    PR_NUMBER = parse(Int, ENV["BUILDKITE_PULL_REQUEST"])
+    TARGET_BRANCH = "remotes/origin/$(BASE_BRANCH)"
+    COMPARE_AGAINST = chomp(read(`git merge-base --fork-point $(TARGET_BRANCH) HEAD`, String))
+
+    exec(`git fetch origin "refs/pull/$(PR_NUMBER)/head:refs/remotes/origin/pr/$(PR_NUMBER)"`)
+
+    #if [[ "$(git show -s --format=%B origin/pr/$(System.PullRequest.PullRequestNumber))" == *"${SKIP_BUILD_COOKIE}"* ]]; then
+    #   SKIP_BUILD="true"
```

Comment:
@giordano is this for rerunning registration without rebuilding binaries?

Reply:
Yes, this _only_ updates the JLL wrappers, without touching the binaries.  See how it's used in https://github.com/JuliaPackaging/Yggdrasil/blob/428fbb633b693a20bc4ef473adcc62cc8003f567/.ci/register_package.jl#L96-L133

Reply:
ah good to know that we need it in the environment as well!